src/log_manager/leader_log_manager.h

```diff
@@ -28,17 +28,24 @@ class LeaderLogManager final {
 
     ~LeaderLogManager() { timer_manager_->StopTimer(RaftcppConstants::TIMER_PUSH_LOGS); }
 
-    void PullLogs(const NodeID &node_id, int64_t next_log_index);
+    // result for the follower handle push log, the result can be used to update
+    // match_log_indexes_ and committed_log_index_
+    void PullLogs(bool result, const NodeID &node_id, int64_t next_log_index);
 
     void Push(const TermID &term_id,
               const std::shared_ptr<raftcpp::RaftcppRequest> &request);
 
-    void Run();
+    void Run(std::unordered_map<int64_t, LogEntry> &logs, int64_t committedIndex);
 
     void Stop();
 
     int64_t CurrLogIndex() const { return curr_log_index_; }
 
+    // attention to all_log_entries_ may be large, so as far as possible no copy
+    std::unordered_map<int64_t, LogEntry> &Logs() { return all_log_entries_; }
```

Comment:
Pay attention to the concurrent conflict.

When give logs from leader_log_manager to non_leader_log_manager, you should ensure that no more operation will be taken in leader_log_manager.

For example:
There may be two operation named A and B in leader_log_manager to modify the logs, and they are blocked because of the mutex. If you return the logs without any restriction, these two operation will modify the logs after non_leader_log_manager takes logs, this is strange and may cause unexpectable problems.

Reply:
I agree with that it's not good enough that we return an reference anyway.

Reply:
so it need adjust structure of logmanager